src/main/java/games/strategy/triplea/ui/BattleDisplay.java

```diff
@@ -493,17 +493,19 @@ public void actionPerformed(final ActionEvent e) {
             }
             chooserScrollPane = new JScrollPane(chooser);
             final Dimension screenResolution = Toolkit.getDefaultToolkit().getScreenSize();
```

Comment:
A more general thought:
Are we really doing the logic here for cases where the screen is too small?
Shouldn't we just substance let do it's job by "making it just work", meaning we could simply disable a horizontal scrollbar and remove all of this custom logic?

Reply:
I tested this scenario because I thought the same thing (i.e. I simply removed all the code used to set the preferred size).  Unfortunately, the dialog height grows as much as necessary to accommodate all the content.  With enough content, I got it to extend past the bounds of my display, and it was impossible to view the portion that extended below my display (at least on GNOME, where even if using the keyboard to move a window, the window manager doesn't allow the title bar to move past the top of the display).

I don't know if this is a `JOptionPane` issue or a L&F issue, but I'm going to assume the former because I've seen it in other `JOptionPane`s (e.g. when we were displaying the complete Hotmail error message described in #2598).

Reply:
Weird.
By simply looking at the code it really seems to me the preferred height of the jscrollpane is set to its preferred height, because the screen height - some set pixel value should be higher in most cases?

Reply:
Here's the select casualties window on a 1024x768 display with no `setPreferredSize()` code:

![select-casualties-large](https://user-images.githubusercontent.com/4826349/34628669-38c1fc68-f233-11e7-8741-b2076870e4dd.png)

and with the `setPreferredSize()` code:

![select-casualties-large-with-scroll](https://user-images.githubusercontent.com/4826349/34628854-e664c670-f233-11e7-9ef9-6ea6916b4c86.png)

(Side note: the 130 pixels subtracted from the display height to accommodate the taskbar/dock may not be enough.  I wonder if there's a Swing API that returns the _desktop_ size as opposed to the _display_ size.)

Reply:
This [SO answer](https://stackoverflow.com/a/10126428/3900879) has a nice simple solution, but it's only good for a single-monitor system.  It's only [slightly more complicated](https://stackoverflow.com/a/10123912/3900879) for a multi-monitor system.  I'm not even sure how the TripleA UI behaves, in general, on a multi-monitor system.

Reply:
I haven't noticed any difference since I bought myself a second monitor, although I must admit I didn't really did any observations.
Another thought that came to my mind: what if a set maximum height value is being set, so the user will only ever see 5 casualties at once or something like that.
When using such a small height for a window, the screen limitations should really not matter.

Reply:
> what if a set maximum height value is being set, so the user will only ever see 5 casualties at once or something like that.

Yeah, that would work, as well.  However, I'm assuming there was a specific reason Chris coded it to use as much real estate as possible.

Reply:
Why don't we just ask him?
Hey @veqryn, do you know if you had a specific reason to code the logic of the Battle Display Windows this way?

Reply:
@RoiEXLab: It was coded that way because swing sucks and doesn't account for scrollbars. (This may have changed in the last 4 years?)
For example, if the height is not enough to accommodate the content, then a 20-22 width scrollbar will be added to allow you scroll up and down.  Then because a 20-22 pixel width scrollbar was added, suddenly the width is no longer enough, and swing adds another scrollbar so that you can scroll left and right.  So the code was made to give the panel an additional 22 pixels width IF it needed a vertical scrollbar, so that it wouldn't also need a horizontal scrollbar.
Hope that makes sense.

Reply:
@veqryn Definitely makes sense.
Thanks for the clarification!